dask/bytes/utils.py

```diff
@@ -37,6 +37,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     "host": "node", "port": 123, "path": "/mnt/datasets/test.csv",
     "url_query": "q=1", "extra": "value"}
     """
+    urlpath = str(urlpath)  # re, urllib don't support pathlib.Path objects
```

Comment:
`stringify_path`, though maybe the callers should be doing this.

Reply:
cc @dlovell